doc-gen/src/cli.ts

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env node
```

Comment:
If this file is intended to be run with node, should it be a `.js` instead of `.ts`? Or do we want to run it with `ts-node`/`tsx`?

Reply:
this all gets built into js from ts and it's that built file that is used with node. adding this here ensures that the header is preserved after the build